62.CAD/fragment_shader.hlsl

```diff
@@ -86,10 +181,31 @@ float4 main(PSInput input) : SV_TARGET
     // draw with previous geometry's style's color :kek:
     col = lineStyles[mainObjects[mainObjectIdx].styleIdx].color;
     col.w *= float(quantizedAlpha)/255.f;
+#elif defined(NBL_DRAW_ARC_LENGTH)
+    
+    nbl::hlsl::shapes::Quadratic<float> quadratic = input.getQuadratic();
+    QuadBezierAnalyticArcLengthCalculator<float> preCompValues_calculator = input.getPrecomputedArcLenData();
+    nbl::hlsl::shapes::Quadratic<float>::ArcLengthPrecomputedValues preCompValues;
+    preCompValues.lenA2 = preCompValues_calculator.lenA2;
+    preCompValues.AdotB = preCompValues_calculator.AdotB;
+    preCompValues.a = preCompValues_calculator.a;
+    preCompValues.b = preCompValues_calculator.b;
+    preCompValues.c = preCompValues_calculator.c;
+    preCompValues.b_over_4a = preCompValues_calculator.b_over_4a;
+    
+    float tA = quadratic.ud(input.position.xy).y;
+
+    float bezierCurveArcLen = quadratic.calcArcLen(1.0, preCompValues);
+    float arcLen = quadratic.calcArcLen(tA, preCompValues);
+    
+    float resultColorIntensity = quadratic.calcArcLenInverse(arcLen, 0.000001f, arcLen / bezierCurveArcLen, preCompValues);
+
+    col = float4(0.0f, resultColorIntensity, 0.0f, 1.0f);
+    
```

Comment:
remove this whole thing

Reply:
will need it to test analytic arc len computation agains numeric

Reply:
We already decided to go with analytic. am I correct?

Reply:
yes, but i just want to test analytic method against numeric, then i will remove everything related to numeric arc len computation